exchange/accounts/balance.go

```diff
@@ -0,0 +1,169 @@
+package accounts
+
+import (
+	"errors"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/thrasher-corp/gocryptotrader/common"
+	"github.com/thrasher-corp/gocryptotrader/currency"
+	"github.com/thrasher-corp/gocryptotrader/exchanges/asset"
+)
+
+var (
+	errBalanceCurrencyMismatch = errors.New("balance currency does not match update currency")
+	errOutOfSequence           = errors.New("out of sequence")
+	errUpdatedAtIsZero         = errors.New("updatedAt may not be zero")
+)
+
+// Balance contains an exchange currency balance
+type Balance struct {
+	Currency               currency.Code
+	Total                  float64
+	Hold                   float64
+	Free                   float64
+	AvailableWithoutBorrow float64
+	Borrowed               float64
+	UpdatedAt              time.Time
+}
+
+// Change defines incoming balance change on currency holdings
+type Change struct {
+	Account   string
+	AssetType asset.Item
+	Balance   Balance
+}
+
+// balance contains a balance with live updates
+type balance struct {
+	internal Balance
+	m        sync.RWMutex
+}
+
+// CurrencyBalances provides a map of currencies to balances
+type CurrencyBalances map[currency.Code]Balance
+
+// currencyBalances provides a map of currencies to balances
+type currencyBalances map[currency.Code]*balance
```

Comment:
```suggestion
type currencyBalances map[*currency.Item]*balance
```

reduce chance that there is a upper/lower case duplication